python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -455,6 +518,55 @@ def resolve_latestMaterializationByPartition(
             for event in ordered_materializations
         ]
 
+    def resolve_materializationCountByDimensionPartition(
+        self, graphene_info, **kwargs
+    ) -> Sequence[GraphenePrimaryPartitionDimensionMaterializationCount]:
+        primary_dimension = kwargs.get("primaryDimension")
+
+        if primary_dimension is None:
+            primary_dimension = self.get_default_primary_dimension_name()
+
+        partition_keys_by_dimension = self.get_partition_keys_by_dimension()
+        secondary_dimension = next(
+            iter(list(set(partition_keys_by_dimension.keys()) - {primary_dimension}))
+        )
+
+        # This dict will by keyed by the primary dimension partition keys.
+        # The values are dicts keyed by the secondary dimension partition keys, mapped to
+        # the number of materializations.
+        materialization_ct: Dict[
+            str, Dict[str, int]
+        ] = get_materialization_ct_by_dimension_partition_keys(
+            graphene_info,
+            self._external_asset_node.asset_key,
+            primary_dimension,
+            secondary_dimension,
+        )
+
+        partitions_materialized_by_dimension: List[
+            GraphenePrimaryPartitionDimensionMaterializationCount
+        ] = []
+
+        check.invariant(primary_dimension in partition_keys_by_dimension)
+        for primary_partition_key in partition_keys_by_dimension[primary_dimension]:
+            secondary_dim_materialization_cts = []
+            for secondary_partition_key in partition_keys_by_dimension[secondary_dimension]:
+                secondary_dim_materialization_cts.append(
+                    GrapheneMaterializationCount(
+                        partition=secondary_partition_key,
```

Comment:
Edit: It's possible we could achieve this with your existing implementation by not asking for the keys in the secondary partition resolver, but I wonder if we could make the query faster by flattening it to a `[Int!]!` in GraphQL instead of an object type

Reply:
I was able to make the response a 2D array!

Though, we decided to consolidate the the two materialization count resolvers into one, and a consequence of that is that each possible return type must be an object. If the query ends up being too slow let me know and we'll try to speed it up